sphinx/builders/latex/transforms.py

```diff
@@ -94,13 +94,16 @@ def expand_show_urls(self) -> None:
     def get_docname_for_node(self, node: Node) -> str:
         while node:
             if isinstance(node, nodes.document):
-                return self.env.path2doc(node['source'])
+                doc = self.env.path2doc(node['source'])
+                if doc is None:
+                    raise ValueError("unreachable code")
+                return doc
```

Comment:
@tk0miya can you please check you're happy with this approach here? is `path2doc` genuinely *never* `None` in this situation?

perhaps there's a better way to refactor this. this doesn't feel very 'clean' as is